### PR TITLE
HOSTEDCP-934: Validate PublishingStrategyMapping

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:
Validates and reports conflicting services published hostnames

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #[HOSTEDCP-934](https://issues.redhat.com/browse/HOSTEDCP-934)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [x] This change includes unit tests.

### CODE REVIEW

**Review by bryan-cox — COMMENTED** [review]
A couple minor comments

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go [hunk 1]
```diff
@@ -3529,18 +3534,41 @@ func (r *HostedClusterReconciler) validateAWSConfig(hc *hyperv1.HostedCluster) e
...
+	hostnameServiceMap := make(map[string]string, len(hc.Spec.Services))
+	for _, svc := range hc.Spec.Services {
+		hostname := ""
+		if svc.Type == hyperv1.LoadBalancer && svc.LoadBalancer != nil {
+			hostname = svc.LoadBalancer.Hostname
+		} else if svc.Type == hyperv1.Route && svc.Route != nil {
+			hostname = svc.Route.Hostname
+		}
+
+		if hostname != "" {
```
Comment (sjenning) [inline hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:3559]: if you invert this
```
if hostname == "" {
  continue
}
```
you can unindent the remaining lines and make it easy to understand i.e. if `hostname` is not specified, this time through the loop is a no-op.

**Review by bryan-cox — APPROVED** [review]
/lgtm

**Comment by enxebre** [comment]
very nice, thanks!
/approve

**Pull request merged**